src/actions/mintArveaweNFT.ts

```diff
@@ -0,0 +1,266 @@
+import { sendTransaction } from './transactions';
+import { updateMetadata } from './updateMetadata';
+import { createMasterEdition } from './createMasterEdition';
+
+export enum ENftProgress {
+  none,
+  minting,
+  preparing_assets,
+  signing_metadata_transaction,
+  sending_transaction_to_solana,
+  waiting_for_initial_confirmation,
+  waiting_for_final_confirmation,
+  uploading_to_arweave,
+  updating_metadata,
+  signing_token_transaction,
+}
+
+export async function convertFile2Hash(file: File): Promise<Buffer> {
+  const hashSum = crypto.createHash('sha256');
+  const data = await file.text();
+  hashSum.update(data);
+  const hex = hashSum.digest('hex');
+  return Buffer.from(hex);
+}
+
+export function getAssetCostInfo(files: Map<string, Buffer>) {
+  const sizes = Array.from(files.values()).map((f) => f.byteLength);
+  return calculate(sizes);
+}
+
+export function createFilePack(metadata: MetadataJson, WebFile = File): File {
+  const filedata = {
+    animation_url: undefined,
+    attributes: metadata.attributes && metadata.attributes.length ? metadata.attributes : undefined,
+
+    name: metadata.name,
+    symbol: metadata.symbol,
+    description: metadata.description,
+    seller_fee_basis_points: metadata.seller_fee_basis_points,
+
+    image: metadata.image,
+    external_url: metadata.external_url,
+    properties: {
+      files: metadata.properties.files,
+      category: metadata.properties.category,
+      creators: metadata.properties.creators.map(({ address, share }) => ({
+        address,
+        share,
+      })),
+    },
+  };
+
+  return new WebFile([JSON.stringify(filedata)], 'metadata.json');
+}
+
+export async function getCostInfo(
+  file: File,
+  metadata: MetadataJson,
+  fileMetadata = createFilePack(metadata),
+) {
+  const [dataFile, dataMetadata] = await Promise.all([
+    file.arrayBuffer().then((d) => Buffer.from(d)),
+    fileMetadata.arrayBuffer().then((d) => Buffer.from(d)),
+  ]);
+  const files = new Map<string, Buffer>();
+  files.set(file.name, dataFile);
+  files.set(fileMetadata.name, dataMetadata);
+  return getAssetCostInfo(files);
+}
+export async function getCost(
+  file: File,
+  metadata: MetadataJson,
+  fileMetadata = createFilePack(metadata),
+) {
+  const calcInfo = await getCostInfo(file, metadata, fileMetadata);
+  return calcInfo.solana * LAMPORTS_PER_SOL;
+}
+
+const ARWEAVE_WALLET = new PublicKey('6FKvsq4ydWFci6nGq9ckbjYMtnmaqAoatz5c9XWjiDuS');
+
+export async function payForFiles(
+  {
+    file,
+    metadata,
+    wallet: w,
+  }: {
+    wallet: Wallet;
+    file: File;
+    metadata: MetadataJson;
+  },
+  WebFile = File,
+) {
+  const fileMetadata = createFilePack(metadata, WebFile);
+  const [hashFile, hashMetadata] = await Promise.all([
+    convertFile2Hash(file),
+    convertFile2Hash(fileMetadata),
+  ]);
+  const files = new Map<string, File>();
+  files.set(file.name, file);
+  files.set(fileMetadata.name, fileMetadata);
```

Comment:
you have similar code in the line 72

Reply:
Not exactly the same. On line 72 `Map<string, Buffer>` and here `Map<string, File>`

Reply:
Not exactly

Reply:
In this case you can use generic type